src/compressed_tensors/quantization/quant_args.py

```diff
@@ -381,28 +392,47 @@ def get_observer(self) -> str:
     model_config = ConfigDict(extra="forbid")
 
 
+def _round_dtype(tensor: torch.Tensor, dtype: torch.dtype):
+    if torch.is_floating_point(torch.tensor([], dtype=dtype)):
+        finfo = torch.finfo(dtype)
+        return torch.clamp(tensor, finfo.min, finfo.max).to(dtype)
+    else:
+        iinfo = torch.iinfo(dtype)
+        return torch.round(torch.clamp(tensor, iinfo.min, iinfo.max))
+
+
+def _round_args(tensor: torch.Tensor, args: QuantizationArgs):
```

Comment:
Why don't we need to clamp in this case? Maybe there's a way to combine this with _round_dtype?

Reply:
It's very strange that this function has different behavior depending on if you pass a dtype or args. It seems like we should fold the clamp you linked above into this function.

Reply:
I guess what you're trying to say is that "the way clamping is done is different in calibration than it is in forward", which is true. As a follow up, we can combine the two concepts, putting together a pr for what that might look like...

Reply:
https://github.com/vllm-project/compressed-tensors/pull/512/files#diff-a1e6707ddccc2ae85a0651c623ea596212a0f5c1bd32b22c05713822956f0d54R435

If you, instead of passing `dtype` or `args`, pass `finfo` or `iinfo`, the function becomes a lot cleaner (and self-evidently safer)